Junimatic/Patcher.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using HarmonyLib;
+using StardewValley;
+
+using static NermNermNerm.Stardew.LocalizeFromSource.SdvLocalize;
+
+namespace NermNermNerm.Junimatic
+{
+    internal static class Patcher
+    {
+        /// <summary>
+        /// Apply the harmony patches for Junimatic
+        /// </summary>
+        internal static void Apply()
+        {
+            ModEntry.Instance.Harmony.Patch(
+                original: AccessTools.Method(typeof(Farmer), nameof(Farmer.addItemToInventoryBool)),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Farmer_addItemToInventoryBool_Prefix))
+            );
+            ModEntry.Instance.Harmony.Patch(
+                original: AccessTools.Method(typeof(Game1), nameof(Game1.createItemDebris)),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Game1_createItemDebris_Prefix))
+            );
+            ModEntry.Instance.Harmony.Patch(
+                original: AccessTools.Method(typeof(Game1), nameof(Game1.createObjectDebris), [typeof(string), typeof(int), typeof(int), typeof(int), typeof(int), typeof(float), typeof(GameLocation)]),
+                prefix: new HarmonyMethod(typeof(Patcher), nameof(Game1_createObjectDebris_Prefix))
+            );
+        }
+
+        internal static List<Item> InterceptHarvest(Action harvestAction, GameLocation machineLocation)
+        {
+            var playerField = typeof(Game1).GetField(I("_player"), System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Static);
+            var player = Game1.player;
+            try
+            {
+                Patcher.collectedItems = new List<Item>();
+
+                // Set the backing field for Game1.player so that calls to do animation on the farmer don't happen and
+                //  farmer's experience isn't used to calculate the quality of the crops and no xp is gained.
+                playerField!.SetValue(null, Patcher.fakePlayer);
+                Patcher.fakePlayer.currentLocation = machineLocation;
+
+                harvestAction();
```

Comment:
Have you tested this with bushes? I needed to set the fakePlayer's daysPlayed stat or it would never harvest (I borderline wrote an essay in my comments so I wouldn't forget that little detail). Does setting _player directly avoid this/allow syncing to set that stat so harvesting bushes works?

Reply:
No I have not, and no, I don't think it would solve the problem at all!  Now that you remind me, I should have copypasted your essay and the associated hack!

(I have tested this with ordinary crops over a few days of playtime.  But haven't tested it with Tea bushes or foragables.)

Reply:
I have no doubt my essay could be condensed. I'll test before proposing any changes. Unfortunately, how stats are shared between players is a bit of a blackbox even to the veterans on the discord (at least based on my interpretation of where that conversation went), so I can't say off hand if creating a persisting fake farmer that is set to game1._player at some point will allow stats to be synced.

Reply:
Yup, you are right.  I updated the code to include the DaysPlayed patch and condensed the essay.